tests/openssl/CMakeLists.txt

```diff
@@ -0,0 +1,512 @@
+    add_openssl_test("${name}-int0" "${name}" "ASN1_INTEGER" "OK"
+                     ${OPENSSL_TEST_DIR}/d2i-tests/int0.der)
+    add_openssl_test("${name}-int1" "${name}" "ASN1_INTEGER" "OK"
+                     ${OPENSSL_TEST_DIR}/d2i-tests/int1.der)
+    add_openssl_test("${name}-intminus1" "${name}" "ASN1_INTEGER" "OK"
+                     ${OPENSSL_TEST_DIR}/d2i-tests/intminus1.der)
+    add_openssl_test("${name}-int0-any" "${name}" "ASN1_ANY" "OK"
+                     ${OPENSSL_TEST_DIR}/d2i-tests/int0.der)
+    add_openssl_test("${name}-int1-any" "${name}" "ASN1_ANY" "OK"
+                     ${OPENSSL_TEST_DIR}/d2i-tests/int1.der)
+    add_openssl_test("${name}-intminus1-any" "${name}" "ASN1_ANY" "OK"
+                     ${OPENSSL_TEST_DIR}/d2i-tests/intminus1.der)
+    add_openssl_test("${name}-bad-int-pad0" "${name}" "ASN1_INTEGER" "decode"
+                     ${OPENSSL_TEST_DIR}/d2i-tests/bad-int-pad0.der)
+    add_openssl_test(
+      "${name}-bad-int-padminus1" "${name}" "ASN1_INTEGER" "decode"
+      ${OPENSSL_TEST_DIR}/d2i-tests/bad-int-padminus1.der)
+    continue()
+  endif ()
+
+  if ("${name}" STREQUAL danetest)
+    add_openssl_test(
+      "${name}" "${name}" "example.com" ${OPENSSL_TEST_DIR}/danetest.pem
+      ${OPENSSL_TEST_DIR}/danetest.in)
+    continue()
+  endif ()
+
+  if ("${name}" STREQUAL dtlstest)
+    add_openssl_test("${name}" "${name}" ${OPENSSL_APPS_DIR}/server.pem
+                     ${OPENSSL_APPS_DIR}/server.pem)
+    continue()
+  endif ()
+
+  if ("${name}" STREQUAL ectest)
+    add_openssl_test("${name}" "${name}")
+    continue()
+  endif ()
+
+  if ("${name}" STREQUAL evp_test)
+    # Disable the tests if the target openssl_test_data is not present.
+    if (NOT TARGET openssl_test_data)
+      continue()
+    endif ()
+
+    add_openssl_test("${name}-evpciph" "${name}"
+                     ${TEST_DATA_DIR}/30-test_evp_data/evpciph.txt)
+    add_openssl_test("${name}-evpdigest" "${name}"
+                     ${TEST_DATA_DIR}/30-test_evp_data/evpdigest.txt)
+    add_openssl_test("${name}-evpencod" "${name}"
+                     ${TEST_DATA_DIR}/30-test_evp_data/evpencod.txt)
+    add_openssl_test("${name}-evpmac" "${name}"
+                     ${TEST_DATA_DIR}/30-test_evp_data/evpmac.txt)
+    add_openssl_test("${name}-evpcase" "${name}"
+                     ${TEST_DATA_DIR}/30-test_evp_data/evpcase.txt)
+    add_openssl_test("${name}-evpccmcavs" "${name}"
+                     ${TEST_DATA_DIR}/30-test_evp_data/evpccmcavs.txt)
+    # The following two tests require sufficient heap memory (e.g., 6144 pages).
+    add_openssl_test("${name}-evpkdf" "${name}"
+                     ${TEST_DATA_DIR}/30-test_evp_data/evpkdf.txt)
+    add_openssl_test("${name}-evppbe" "${name}"
+                     ${TEST_DATA_DIR}/30-test_evp_data/evppbe.txt)
+    # This test requires insufficient heap memory (e.g., 6144 pages) with DEBUG MALLOC.
```

Comment:
@nonpolarity  We need to enable these tests with DEBUG_MALLOC by via the runtime configuration.